valhalla/worker.h

```diff
@@ -74,6 +75,26 @@ void ParseApi(const prime_server::http_request_t& http_request, Api& api);
 #endif
 
 std::string serialize_error(const valhalla_exception_t& exception, Api& options);
+
+// unordered map for warning pairs
+const std::unordered_map<int, std::string> warnings_object = {
+    {100, "auto_shorter is deprecated and will be turned into the shotest costing option"},
+    {101,
+     "hov costing is deprecated and will be turned into auto costing with hov2=true costing option"},
+    {102, "auto_data_fix is deprecated and will be turned to ignore all the things costing option"},
+    {103, "best_paths has been deprecated. use alternates instead"},
+};
+
+// function to add warnings to proto info object
+inline void add_warning(valhalla::Api& api, unsigned code) {
+  auto message = warnings_object.find(code);
+  if (message != warnings_object.end()) {
+    auto* warning = api.mutable_info()->mutable_warnings()->Add();
+    warning->set_description(message->second);
+    warning->set_code(message->first);
+  }
+}
```

Comment:
I think this should all go to the .cc file. well, except the `add_warning` definition, but the function body should move as well. the warning codes shouldn't be part of the public api, they can stay "private", just like the error codes

Reply:
yeah, the signature of the function should be in the header file. but the implementation (IMHO) should be in the .cc file. the reason is that with the above the warning codes are in `valhalla` namespace and "pollute" the public api. if another project uses valhalla as a library they have no use for those codes. in order for them not to be seen we put them in an anonymous namespace (and they're just dead weight in a header in my experience). IMHO it's also nicer style to keep the header files as slim/clean as reasonable to immediately see what it's containing.

it's not a big deal, bit nit picky, but there's at least actual reasons;)

Reply:
yep the `declaration` should go in the header and the `definition` or `implementation` should go in the source file